### DISCUSSION TITLE
Visual mode (VSCode extension) could be more flexible about the spell-checking language(s).

### DISCUSSION THREAD

**Discussion by fuhrmanator** [quarto-dev/quarto-cli#6774] [Feature Requests]
### Description

I often write documents in either English or French (sometimes with a mix), but it appears that **visual mode** only supports a single language. 

![image](https://github.com/quarto-dev/quarto-cli/assets/7606540/f51e7f2f-2f40-4883-bb88-1049d0056664)

VSCode has various extensions for checking spelling, e.g., Code Spell Checker, and they support many languages. 

So, I am asking for a feature that is more flexible. Perhaps by integrating Code Spell Checker settings, perhaps by using the `lang:` from the YAML of the current document.

Thanks as usual for the Quarto progress! I'm happy to see **visual mode** in VSCode :smile:

**Comment by mcanouil** [comment]
This is for the VSCode Quarto extension repository, see <https://github.com/quarto-dev/quarto>.

Note that I am using Code Spell Checker with Quarto without having to do anything within Quarto CLI or the VSCode extension.

**Reply by fuhrmanator** [reply]
> This is for the VSCode Quarto extension repository, see https://github.com/quarto-dev/quarto.

Thanks for your reply. I'm not sure if you're saying I should create an issue on https://github.com/quarto-dev/quarto? There is no discussion space there for feature requests.

> Note that I am using Code Spell Checker with Quarto without having to do anything within Quarto CLI or the VSCode extension.

Are you using more than one language? I have Code Spell Checker set up for both English and French, which works great for editing code (it doesn't require any switching). However, as soon as I activate the Visual Editor for Quarto, it forces me to use the language setting used by that tool. 

For example, if the Visual Editor language setting is English, it will underline all French words (despite the Code Spell Checker setting that allows both languages). I can change the setting to French, but then any documents I edit in English will underline the English words.

**Reply by mcanouil** [reply]
Oh, the issue is really about the visual editor (which I do not use), so please do open on the repo I linked.
Discussion will happen over there.
Note that the Visual Editor is one thing shared across RStudio and VSCode.
For any feature requests or bug reports regarding editors, they should be made on the corresponding repositories.

**Reply by mcanouil** [reply]
For tracking, please comment with the link of the issue you are going to open for others to find where the discussion is.
Thank you!

**Reply by fuhrmanator** [reply]
I posted to https://github.com/quarto-dev/quarto/issues/265 and mentioned this discussion.
